GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m

```diff
@@ -91,4 +98,40 @@ - (void)testLegacyCoding {
 }
 #endif // TARGET_OS_IOS || TARGET_OS_MACCATALYST
 
+- (void)testUpdateAuthState {
+  NSTimeInterval accessTokenExpireTime = [NSDate timeIntervalSinceReferenceDate];
+  NSTimeInterval idTokenExpireTime = accessTokenExpireTime + kTimeIncrement;
+  
+  NSString *idToken = [self idTokenWithExpireTime:idTokenExpireTime];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithIDToken:idToken
+                                                      accessToken:kAccessToken
+                                            accessTokenExpireTime:accessTokenExpireTime];
+  
+  GIDGoogleUser *user = [[GIDGoogleUser alloc] initWithAuthState:authState profileData:nil];
+  
+  NSTimeInterval updatedAccessTokenExpireTime = idTokenExpireTime + kTimeIncrement;
+  NSTimeInterval updatedIDTokenExpireTime = updatedAccessTokenExpireTime + kTimeIncrement;
+  NSString *updatedIDToken = [self idTokenWithExpireTime:updatedIDTokenExpireTime];
+  OIDAuthState *updatedAuthState = [OIDAuthState testInstanceWithIDToken:updatedIDToken
+                                                             accessToken:kNewAccessToken
+                                                   accessTokenExpireTime:updatedAccessTokenExpireTime];
+  GIDProfileData *updatedProfileData = [GIDProfileData testInstance];
+  
+  [user updateAuthState:updatedAuthState profileData:updatedProfileData];
+  
+  XCTAssertEqualObjects(user.accessToken.tokenString, kNewAccessToken);
+  XCTAssertEqualWithAccuracy([user.accessToken.expirationDate timeIntervalSinceReferenceDate],
+                             updatedAccessTokenExpireTime, kTimeAccuracy);
+  XCTAssertEqualObjects(user.idToken.tokenString, updatedIDToken);
+  XCTAssertEqualWithAccuracy([user.idToken.expirationDate timeIntervalSinceReferenceDate],
+                             updatedIDTokenExpireTime, kTimeAccuracy);
+  XCTAssertEqual(user.profile, updatedProfileData);
+}
+
+#pragma mark - Helpers
+
+- (NSString *)idTokenWithExpireTime:(NSTimeInterval)expireTime {
+  return [OIDTokenResponse idTokenWithSub:kUserID exp:@(expireTime + NSTimeIntervalSince1970)];
```

Comment:
Why does this method use `expireTime + NSTimeIntervalSince1970`? Removing `NSTimeIntervalSince1970` causes the tests to fail, which is confusing to me. 

Also, this method name suggests that it makes an ID token with the given `expireTime`, but what it does is create an ID token with the given expire time plus this extra increment. Maybe update the name and add a comment to clarify what it is doing and why.

Reply:
This is the method we used to generate an id token in GIDAuthenticationTest. We can figure it out how it works offline and improve it in a separate PR.

Reply:
> Why does this method use `expireTime + NSTimeIntervalSince1970`? Removing `NSTimeIntervalSince1970` causes the tests to fail, which is confusing to me.

The `NSTimeInterval` values generated by [`timeIntervalSinceReferenceDate`](https://developer.apple.com/documentation/foundation/nsdate/1408189-timeintervalsincereferencedate?language=objc) are relative to the reference date of Jan 1, 2001.  Here, [`NSTimeIntervalSince1970`](https://developer.apple.com/documentation/foundation/nstimeintervalsince1970?language=objc) is being used to convert it to a unix epoch value, which is the reference date used by the tokens.  We could probably simplify by using [`timeIntervalSince1970`](https://developer.apple.com/documentation/foundation/nsdate/1407504-timeintervalsince1970?language=objc) instead.

Reply:
Thanks for clarifying. I had a feeling that was the case. I agree that simplifying would be better